healthcareai/supervised_model_trainer.py

```diff
@@ -47,6 +47,26 @@ def __init__(self, dataframe, predicted_column, model_type, impute=True, grain_c
         # Split the data into train and test
         self._advanced_trainer.train_test_split()
 
+        # Save the original dataframe columns to the parent class
+        self._advanced_trainer.pre_dummified_columns = dataframe.columns.values
```

Comment:
This is a little tricky for a few reasons.
1. This takes the original dataframe after the pipeline has been run, which drops some columns (DTS, grain columns, etc), and we don't necessarily want to enforce those being there for predictions to work.
2. I would prefer if these original column values are an optional constructor argument so it is more clear to users how to use the class, rather than setting a property after the fact.